scripts/idl/test_matter_idl_parser.py

```diff
@@ -79,12 +79,35 @@ def test_global_struct(self):
                    fields=[
                         Field(
                             data_type=DataType(name="CHAR_STRING"), code=1, name="astring", ),
-                        Field(data_type=DataType(name="CLUSTER_ID"), code=2, name="idlist", is_list=True, attributes=set(
-                            [FieldAttribute.OPTIONAL])),
-                        Field(data_type=DataType(name="int"), code=0x123, name="valueThatIsNullable", attributes=set(
-                            [FieldAttribute.NULLABLE])),
+                        Field(data_type=DataType(name="CLUSTER_ID"), code=2, name="idlist", is_list=True, qualities=set(
+                            [FieldQuality.OPTIONAL])),
+                        Field(data_type=DataType(name="int"), code=0x123, name="valueThatIsNullable", qualities=set(
+                            [FieldQuality.NULLABLE])),
                         Field(data_type=DataType(name="char_string", max_length=123),
-                              code=222, name="sized_string", attributes=set()),
+                              code=222, name="sized_string", qualities=set()),
+                   ])]
+        )
+        self.assertEqual(actual, expected)
+
+    def test_fabric_scoped_struct(self):
+        actual = parseText("""
+            fabric_scoped struct FabricStruct {
+                CHAR_STRING astring = 1;
+                optional CLUSTER_ID idlist[] = 2;
+                nullable fabric_sensitive int nullablesensitive = 0x123;
```

Comment:
Do you have a negative test for having fields tagged `fabric_sensitive` if they are not inside a fabric-scoped struct?

Reply:
We do not detect such a thing at all, so I guess now. We currently just copy over flags without validation.

Reply:
I believe more linting logic should be separate PRs (like we had a suggestion to detect fabric_idx existence for fabric_scoped structures).